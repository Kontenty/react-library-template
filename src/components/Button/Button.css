.button {
  font-size: 16px;
  border-radius: 5px;
  min-width: 150px;
  height: 38px;
  line-height: 38px;
  padding-right: 1rem;
  padding-left: 1rem;
  border: none;
  cursor: pointer;
}

.primary {
  background-color: #1f51ff;
  color: white;
}
.secondary {
  background-color: hotpink;
  color: white;
}
