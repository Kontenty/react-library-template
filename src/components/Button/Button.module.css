button {
  font-size: 16px;
  border-radius: 5px;
  min-width: 150px;
  height: 38px;
  line-height: 38px;
  padding-right: 1rem;
  padding-left: 1rem;
  border: none;
  cursor: pointer;
}

button.primary {
  background-color: #1f51ff;
  color: white;
}
button.secondary {
  background-color: hotpink;
  color: white;
}
