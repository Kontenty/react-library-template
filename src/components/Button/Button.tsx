import React from "react";
import type { ReactNode } from "react";
import clsn from "classnames";
import css from "./Button.module.css";

type Props = { children: ReactNode; variant?: "primary" | "secondary" };

const Button = ({ children, variant = "primary" }: Props) => {
  return (
    <button type="button" className={clsn(css.button, css[variant])}>
      {children}
    </button>
  );
};

export default Button;
