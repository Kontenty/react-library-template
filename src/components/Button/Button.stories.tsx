import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import Button from "./Button";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const Meta: ComponentMeta<typeof Button> = {
  title: "Components/Button",
  component: Button,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    // backgroundColor: { control: 'color' },
  },
};
export default Meta;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Button> = ({ children, ...args }) => (
  <Button {...args}>{children}</Button>
);

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Primary.args = {
  variant: "primary",
  children: "Button",
};

export const Secondary = Template.bind({});
Secondary.args = {
  variant: "secondary",
  children: "Button",
};
