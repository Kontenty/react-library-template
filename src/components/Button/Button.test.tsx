import React from "react";
import { render } from "@testing-library/react";

import Button from "./Button";

describe("Button", () => {
  test("renders Button component", () => {
    render(<Button>example button</Button>);
  });

  it("styles primary button appropriately", () => {
    const container = render(<Button variant="primary">Primary</Button>);
    const button = container.getByRole("button");
    expect(button).toMatchSnapshot();
  });

  it("styles secondary button appropriately", () => {
    const container = render(<Button variant="secondary">Secondary</Button>);
    const button = container.getByRole("button");
    expect(button).toMatchSnapshot();
  });
});
