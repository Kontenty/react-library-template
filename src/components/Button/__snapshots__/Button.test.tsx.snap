// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button styles primary button appropriately 1`] = `
<button
  class="button primary"
  type="button"
>
  Primary
</button>
`;

exports[`Button styles secondary button appropriately 1`] = `
<button
  class="button secondary"
  type="button"
>
  Secondary
</button>
`;
